src/service/resources-impl.js

```diff
@@ -629,6 +629,14 @@ export class ResourcesImpl {
       // TODO(willchou): Fix observe/unobserve/remeasure churn in reparenting.
       this.intersectionObserver_.unobserve(resource.element);
     }
+
+    if (resource.getState() === ResourceState.LAYOUT_SCHEDULED) {
+      // TODO: should we call resource.unload() instead?
+      // Has the same effect but much stranger codepath.
```

Comment:
I'm not sure when this PR changed code, but if we're doing this you should be calling `r.layoutCanceled()`.

Reply:
I rerequested review because I realized my prior fix (of making cleanupTasks more accepting) could have caused unintended bugs. I think this one will be safer

Reply:
great! that function does _almost_ exactly what I want. It has a bug for `intersect-resources` though :/